src/parameters.rs

```diff
@@ -74,76 +77,53 @@ pub struct SystemParameters {
     pub G_a1:      RistrettoPoint,
 }
 
-macro_rules! try_deserialise {
-    ($name:expr, $bytes:expr) => {
-        match CompressedRistretto($bytes).decompress() {
-            Some(x)  => x,
-            None     => {
-                #[cfg(feature = "std")]
-                println!("Could not decode {:?} from bytes: {:?}", $name, $bytes);
+macro_rules! try_deserialize {
+    ($name:expr, $bytes:expr) => {{
+        let bytes = $bytes;
+        match CompressedRistretto::from_slice(&bytes).decompress() {
+            Some(x) => x,
+            None => {
+                #[cfg(all(debug_assertions, feature = "std", feature = "debug-errors"))]
+                eprintln!("Could not decode {:?} from bytes: {:?}", $name, bytes);
```

Comment:
I didn't add the `debug-errors` feature to `Cargo.toml` but I'm not sure this code belongs here.